java-checks/src/main/java/org/sonar/java/symexec/SymbolicEvaluator.java

```diff
@@ -661,92 +701,47 @@ public void visitWhileStatement(WhileStatementTree tree) {
 
     void invalidateAssignedVariables(Set<Symbol.VariableSymbol> assignedVariables) {
       for (Symbol.VariableSymbol symbol : assignedVariables) {
-        for (ExecutionState state : currentStates) {
-          state.setBooleanConstraint(new SymbolicValue.SymbolicVariableValue(symbol), SymbolicBooleanConstraint.UNKNOWN);
+        SymbolicValue value = new SymbolicValue();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this instantiation of "SymbolicValue". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2440)